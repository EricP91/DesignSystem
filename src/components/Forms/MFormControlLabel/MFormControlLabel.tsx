import React from 'react';
import { FormControlLabel, FormControlLabelProps, Typography } from '@mui/material';

export type MFormControlLabelProps = FormControlLabelProps;

const MFormControlLabel = ({ label, disabled, ...props }: MFormControlLabelProps): JSX.Element => {
  const isLabelNumberOrString = typeof label === 'string' || typeof label === 'number';

  return (
    <FormControlLabel
      label={isLabelNumberOrString ? <Typography variant="text">{label}</Typography> : label}
      {...props}
    />
  );
};

export default MFormControlLabel;
