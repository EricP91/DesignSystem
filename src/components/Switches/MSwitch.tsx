import React from 'react';
import Switch from '@mui/material/Switch';
import { SwitchProps } from '@mui/material';

export type MSwitchProps = SwitchProps;

function MSwitch({ ...props }: MSwitchProps): JSX.Element {
  return <Switch disableRipple {...props} />;
}

export default MSwitch;
