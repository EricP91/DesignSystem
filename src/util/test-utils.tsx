import React, { FC, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeConfig } from '../theme';

const AllTheProviders: FC = ({ children }): JSX.Element => <ThemeConfig isLightMode>{children}</ThemeConfig>;

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
