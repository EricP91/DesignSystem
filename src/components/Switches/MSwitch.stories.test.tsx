import { MFormControlLabelProps } from 'components/Forms/MFormControlLabel/MFormControlLabel';
import React from 'react';

import { render } from 'test-utils';
import { Default } from './MSwitch.stories';

test('should render MSwitch story', () => {
  const story = render(<Default {...(Default.args as MFormControlLabelProps)} />);
  expect(story).toMatchSnapshot();
});
