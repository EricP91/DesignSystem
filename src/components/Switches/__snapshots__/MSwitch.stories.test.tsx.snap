// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MSwitch story 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-17av8be-MuiFormControlLabel-root"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeMedium css-f201mg-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-aa25ke-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-zy2el9-MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiSwitch-track css-jsuk3q-MuiSwitch-track"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-text css-1me7gdt-MuiTypography-root"
        >
          Switch title
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-17av8be-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-f201mg-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-aa25ke-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-zy2el9-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-jsuk3q-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-text css-1me7gdt-MuiTypography-root"
      >
        Switch title
      </span>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
